### PR TITLE
Fix BankOfCorda demo after services removal

### PR DESCRIPTION
**Pull request by kasiastreich**
Remove DUMMY_NOTARY.name because it causes confusion and is source of potential inconsistency between configs/notary identity created for a node

### CODE REVIEW

**Comment by joeldudleyr3** [comment]
To use this from the explorer, we also need to add the `"StartFlow.net.corda.finance.flows.CashConfigDataFlow"` permission to `BankOfCorda` in the build.gradle file.

**Comment by joeldudleyr3** [comment]
Explorer is failing with the stacktrace below, so I can't test the new permission. Is this happening for everyone else?

```
Exception in Application constructor
java.lang.reflect.InvocationTargetException
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:498)
        at sun.launcher.LauncherHelper$FXHelper.main(LauncherHelper.java:767)
Caused by: java.lang.RuntimeException: Unable to construct Application instance: class net.corda.explorer.Main
        at com.sun.javafx.application.LauncherImpl.launchApplication1(LauncherImpl.java:907)
        at com.sun.javafx.application.LauncherImpl.lambda$launchApplication$155(LauncherImpl.java:182)
        at java.lang.Thread.run(Thread.java:748)
Caused by: java.lang.reflect.InvocationTargetException
        at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
        at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
        at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
        at java.lang.reflect.Constructor.newInstance(Constructor.java:423)
        at com.sun.javafx.application.LauncherImpl.lambda$launchApplication1$161(LauncherImpl.java:819)
        at com.sun.javafx.application.PlatformImpl.lambda$runAndWait$175(PlatformImpl.java:326)
        at com.sun.javafx.application.PlatformImpl.lambda$null$173(PlatformImpl.java:295)
        at java.security.AccessController.doPrivileged(Native Method)
        at com.sun.javafx.application.PlatformImpl.lambda$runLater$174(PlatformImpl.java:294)
        at com.sun.glass.ui.InvokeLaterDispatcher$Future.run(InvokeLaterDispatcher.java:95)
Caused by: java.lang.NullPointerException
        at javafx.collections.FXCollections.observableList(FXCollections.java:98)
        at net.corda.client.jfx.model.NetworkIdentityModel.<init>(NetworkIdentityModel.kt:40)
        at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
        at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
        at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
        at java.lang.reflect.Constructor.newInstance(Constructor.java:423)
        at java.lang.Class.newInstance(Class.java:442)
        at net.corda.client.jfx.model.Models.initModel(Models.kt:74)
        at net.corda.client.jfx.model.TrackedDelegate.<init>(TrackedDelegate.kt:17)
        at net.corda.client.jfx.model.TrackedDelegate.<init>(TrackedDelegate.kt:15)
        at net.corda.client.jfx.model.TrackedDelegate$ObservableValueDelegate.<init>(TrackedDelegate.kt:50)
        at net.corda.explorer.views.TransactionViewer.<init>(TransactionViewer.kt:367)
        at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
        at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
        at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
        at java.lang.reflect.Constructor.newInstance(Constructor.java:423)
        at java.lang.Class.newInstance(Class.java:442)
        at tornadofx.FXKt.find(FX.kt:295)
        at net.corda.explorer.Main.<init>(Main.kt:145)
        ... 10 more
```

**Comment by kasiastreich** [comment]
Yes, explorer fix is not merged yet into master

File: samples/bank-of-corda-demo/src/main/kotlin/net/corda/bank/BankOfCordaDriver.kt [hunk 1]
```diff
@@ -54,7 +54,7 @@ private class BankOfCordaDriver {
         // The ISSUE_CASH will request some Cash from the ISSUER on behalf of Big Corporation node
         val role = options.valueOf(roleArg)!!
 
-        val requestParams = IssueRequestParams(options.valueOf(quantity), options.valueOf(currency), BIGCORP_LEGAL_NAME, "1", BOC.name, DUMMY_NOTARY.name)
```
Comment (rnicoll) [inline samples/bank-of-corda-demo/src/main/kotlin/net/corda/bank/BankOfCordaDriver.kt]: Can we hard-code the notary service identity, as in "CN=notary.validating,O=Notary Service,L=Zurich,C=CH", and the receiving API then filters notaries to find a match?

File: samples/bank-of-corda-demo/src/main/kotlin/net/corda/bank/api/BankOfCordaClientApi.kt [hunk 2]
```diff
@@ -41,8 +41,7 @@ class BankOfCordaClientApi(val hostAndPort: NetworkHostAndPort) {
             // Resolve parties via RPC
             val issueToParty = rpc.partyFromX500Name(params.issueToPartyName)
                     ?: throw Exception("Unable to locate ${params.issueToPartyName} in Network Map Service")
-            val notaryLegalIdentity = rpc.partyFromX500Name(params.notaryName)
-                    ?: throw IllegalStateException("Unable to locate ${params.notaryName} in Network Map Service")
+            val notaryLegalIdentity = rpc.notaryIdentities().firstOrNull() ?: throw Exception("Couldn't locate any notary in NetworkMapCache")
```
Comment (rnicoll) [inline samples/bank-of-corda-demo/src/main/kotlin/net/corda/bank/api/BankOfCordaClientApi.kt]: Oops - the above example `issueToParty` line should be `IllegalStateException`, and then please update this to match that .

File: samples/bank-of-corda-demo/src/main/kotlin/net/corda/bank/BankOfCordaDriver.kt (hunk 1 above)
Comment (m4ksio) [reply samples/bank-of-corda-demo/src/main/kotlin/net/corda/bank/BankOfCordaDriver.kt]: It opens possible problem is this changes, or somebody makes a typo somewhere

Comment (kasiastreich) [reply samples/bank-of-corda-demo/src/main/kotlin/net/corda/bank/BankOfCordaDriver.kt]: All our samples are like that unfortunately. @rnicoll are you sure we want to hardcode it?

**Comment by kasiastreich** [comment]
Comments addressed, and rebased on current master

**Pull request merged**